5.0 Draft/IDD/IDDs Service Registry/eu.arrowhead.service-discovery-http-json.yml

```diff
@@ -0,0 +1,631 @@
+    messages _must_ be American English (`en-US`).
+
+    ## Managing Record Life-Cycles
+
+    In a typical use case, this service interface is meant to hold records for all service providers
+    currently part of its context. When a service provider joins or leaves a context, it is expected
+    to register and deregister its services, respectively. However, service providers can fail or be
+    detached from the context before they get a chance to deregister their services. This can make
+    it relevant to monitor service providers and temporarily suppress or permanently prune records
+    associated with unhealthy or inaccessible providers. Implementations of this service interface
+    are free to suppress and/or prune records in whatever way best supports the use cases they are
+    designed for.
+
+    ## Record Ownership
+
+    It is the provider indicated inside each record that is the owner of the record, and not the
+    consumer registering it. This means that a consumer may be forbidden from registering or
+    deregistering a record for certain or all other service providers, even if it may register
+    and/or deregister records with itself as provider.
+
+    ## Provider Identifiers
+
+    Every service instance registered via this service interface _must_ be associated with a unique
+    identifier. In contexts where authentication is not used, providers _must_ chose their own
+    identifiers. Those identifiers _should_ remain consistent between restarts of those providers.
+    In contexts where authentication is used, the identifier _must_ be derived as follows: TODO.
+
+    ## Size Limits
+
+    As a mitigation against denial-of-service attacks, all implementations of this service
+    interface _should_ reject incoming requests that are larger than a predefined limit. That limit
+    _must not_ be smaller than 8192 bytes for each received request. An implementation _may_ also
+    limit the sizes of individual parts within each request. It _must_, however, receive request
+    URI:s up to 2816 bytes in size, as well as request payloads up to 4096 bytes in size.
+  version: 5.0.0-rc.1
+  # Additional information present in the IDD template
+  x-transfer-protocol: [HTTP, HTTPS]
+  x-encryption-method: to-add
+  x-encoding: JSON
+  x-compression: to-add
+  x-semantics: to-add
+  x-Ontology: to-add
+
+paths:
+
+  /echo:
+    get:  
+      summary: Endpoint to check the core system availability. 
+      description: Returns an OK to indicate that the core system is available. 
+      responses:
+        '200':
+          description: OK
+
+  /services:
+
+    get:
+      operationId: query
+      description: |
+        Searches for records matching any query parameters. If no query parameters are specified,
+        any records the consuming system are permitted to access may be provided in the response.
+
+        If the `limit` query parameter is used and given a value larger than 16, or if the `offset`
```

Comment:
This introduces unnecessary complexity. Why not offer two different operations then so it is clear? Sometimes something similar to a `POST /services/search` endpoint is offered in APIs.

Reply:
Or maybe `services/query` to keep the naming intact in this case. This would also allow us to define a proper well-defined schema for the query using a request body.

Reply:
How does this avoid complexity? I guess you're not referring to implementation complexity? The implementation will just be something like `if (params.limit > 16) { accessRight = "scan"; } else { accessRight = "query"; }`. I'd argue that adding another endpoint is more complicated.

If you mean that it is confusing or surprising for a human that the same endpoint could require two different access rights, then I can agree. To me, it seems as if both trade-offs are about as good. If you insist I'd say we go with your proposal.